packages/features/bookings/Booker/useSlotReservationId.ts

```diff
@@ -0,0 +1,15 @@
+// TODO: It would be lost on refresh, so we need to persist it.
+// Though, we are persisting it in a cookie(`uid` cookie is set through reserveSlot call)
+// but that becomes a third party cookie in context of embed and thus isn't accessible inside embed
+// So, we need to persist it in top window as first party cookie in that case.
+let slotReservationId: null | string = null;
+
+export const useSlotReservationId = () => {
```

Comment:
I moved the slot reservation Id logic here @vijayraghav-io @Udit-takkar. This is so that we can improve upon this so that it's not lost on refresh.

Reply:
I didn't use useRef because the goal is to keep it persisted as much as we can in memory. Using useRef would cause it to be lost when BookEventForm is unmounted/remounted.